### ISSUE TITLE
[mptcp v0.91] BUG: soft lockup - CPU#3 stuck for 22s! __mptcp_reinject_data

### ISSUE THREAD

**Issue by nadne** [multipath-tcp/mptcp#188]
Hello, 

During a transference with a high number of packets we repeatedly get this error:

```
Jul 12 09:55:23 ip-172-31-23-215 kernel: [763428.208053] NMI watchdog: BUG: soft lockup - CPU#3 stuck for 22s! [test_app:26992]
Jul 12 09:55:23 ip-172-31-23-215 kernel: [763428.208053] Modules linked in: bridge(E) stp(E) llc(E) arptable_filter(E) arp_tables(E) xt_conntrack(E) xt_mark(E) iptable_mangle(E) xt_REDIRECT(E) nf_nat_redirect(E) xt_tcpudp(E) veth(E) iptable_filter(E) ipt_MASQUERADE(E) nf_nat_masquerade_ipv4(E) iptable_nat(E) nf_conntrack_ipv4(E) nf_defrag_ipv4(E) nf_nat_ipv4(E) nf_nat(E) nf_conntrack(E) ip_tables(E) x_tables(E) intel_rapl(E) cirrus(E) ttm(E) crct10dif_pclmul(E) crc32_pclmul(E) drm_kms_helper(E) drm(E) aesni_intel(E) syscopyarea(E) aes_x86_64(E) sysfillrect(E) lrw(E) gf128mul(E) ppdev(E) sysimgblt(E) glue_helper(E) ablk_helper(E) cryptd(E) snd_pcsp(E) evbug(E) 8250_fintek(E) parport_pc(E) snd_pcm(E) parport(E) mac_hid(E) snd_timer(E) snd(E) serio_raw(E) soundcore(E) mptcp_test(E) autofs4(E) ixgbevf(OE) pata_acpi(E)
Jul 12 09:55:23 ip-172-31-23-215 kernel: [763428.208053] CPU: 3 PID: 26992 Comm: test_app Tainted: G           OE   4.1.26test+ #119
Jul 12 09:55:23 ip-172-31-23-215 kernel: [763428.208053] Hardware name: Xen HVM domU, BIOS 4.2.amazon 02/16/2017
Jul 12 09:55:23 ip-172-31-23-215 kernel: [763428.208053] task: ffff8803f50ea7c0 ti: ffff8803cab90000 task.ti: ffff8803cab90000
Jul 12 09:55:23 ip-172-31-23-215 kernel: [763428.208053] RIP: 0010:[<ffffffff817b84e4>]  [<ffffffff817b84e4>] __mptcp_reinject_data+0x164/0x350
Jul 12 09:55:23 ip-172-31-23-215 kernel: [763428.208053] RSP: 0000:ffff8803ff2c3d78  EFLAGS: 00000297
Jul 12 09:55:23 ip-172-31-23-215 kernel: [763428.208053] RAX: ffff880313a19001 RBX: ffffffff816b0cfc RCX: 0000000000cba4e5
Jul 12 09:55:23 ip-172-31-23-215 kernel: [763428.208053] RDX: 0000000000cba4e4 RSI: ffff880313a18e00 RDI: ffffffff816b096a
Jul 12 09:55:23 ip-172-31-23-215 kernel: [763428.208053] RBP: ffff8803ff2c3da8 R08: 000000000001ad20 R09: ffff8803fec03600
Jul 12 09:55:23 ip-172-31-23-215 kernel: [763428.208053] R10: ffff8803ff2dad20 R11: ffffea000c4e8600 R12: ffff8803ff2c3ce8
Jul 12 09:55:23 ip-172-31-23-215 kernel: [763428.208053] R13: ffffffff817edb1e R14: ffff8803ff2c3da8 R15: ffff8803d26b0c00
Jul 12 09:55:23 ip-172-31-23-215 kernel: [763428.208053] FS:  00007faf9aa6d700(0000) GS:ffff8803ff2c0000(0000) knlGS:0000000000000000
Jul 12 09:55:23 ip-172-31-23-215 kernel: [763428.208053] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
Jul 12 09:55:23 ip-172-31-23-215 kernel: [763428.208053] CR2: 00007fea2c6d2162 CR3: 00000000df0df000 CR4: 00000000001406e0
Jul 12 09:55:23 ip-172-31-23-215 kernel: [763428.208053] DR0: 0000000000000000 DR1: 0000000000000000 DR2: 0000000000000000
Jul 12 09:55:23 ip-172-31-23-215 kernel: [763428.208053] DR3: 0000000000000000 DR6: 00000000fffe0ff0 DR7: 0000000000000400
Jul 12 09:55:23 ip-172-31-23-215 kernel: [763428.208053] Stack:
Jul 12 09:55:23 ip-172-31-23-215 kernel: [763428.208053]  ffff8803ec3cd200 ffff8802ebe6e400 0000000000000001 ffff8800eb340a30
Jul 12 09:55:23 ip-172-31-23-215 kernel: [763428.208053]  ffff8800eb340900 ffff8800e2cd5100 ffff8803ff2c3de8 ffffffff817b876b
Jul 12 09:55:23 ip-172-31-23-215 kernel: [763428.208053]  ffff8803ff2c3e28 ffff8800eb340900 ffff8800eb340974 ffff8800e2cd5178
Jul 12 09:55:23 ip-172-31-23-215 kernel: [763428.208053] Call Trace:
Jul 12 09:55:23 ip-172-31-23-215 kernel: [763428.208053]  <IRQ>
Jul 12 09:55:23 ip-172-31-23-215 kernel: [763428.208053]  [<ffffffff817b876b>] mptcp_reinject_data+0x9b/0x150
Jul 12 09:55:23 ip-172-31-23-215 kernel: [763428.208053]  [<ffffffff81726c80>] ? tcp_write_timer_handler+0x210/0x210
Jul 12 09:55:23 ip-172-31-23-215 kernel: [763428.208053]  [<ffffffff817ba5cd>] mptcp_sub_retransmit_timer+0x3d/0xe0
Jul 12 09:55:23 ip-172-31-23-215 kernel: [763428.208053]  [<ffffffff81726b15>] tcp_write_timer_handler+0xa5/0x210
Jul 12 09:55:23 ip-172-31-23-215 kernel: [763428.208053]  [<ffffffff81726d4c>] tcp_write_timer+0xcc/0xd0
Jul 12 09:55:23 ip-172-31-23-215 kernel: [763428.208053]  [<ffffffff810e59b9>] call_timer_fn+0x39/0x160
Jul 12 09:55:23 ip-172-31-23-215 kernel: [763428.208053]  [<ffffffff81726c80>] ? tcp_write_timer_handler+0x210/0x210
Jul 12 09:55:23 ip-172-31-23-215 kernel: [763428.208053]  [<ffffffff810e78d0>] run_timer_softirq+0x260/0x330
Jul 12 09:55:23 ip-172-31-23-215 kernel: [763428.208053]  [<ffffffff8107f607>] __do_softirq+0x117/0x2c0
Jul 12 09:55:23 ip-172-31-23-215 kernel: [763428.208053]  [<ffffffff8107f925>] irq_exit+0x95/0xa0
Jul 12 09:55:23 ip-172-31-23-215 kernel: [763428.208053]  [<ffffffff81489939>] xen_evtchn_do_upcall+0x39/0x50
Jul 12 09:55:23 ip-172-31-23-215 kernel: [763428.208053]  [<ffffffff817edb1e>] xen_hvm_callback_vector+0x6e/0x80
Jul 12 09:55:23 ip-172-31-23-215 kernel: [763428.208053]  <EOI>
Jul 12 09:55:23 ip-172-31-23-215 kernel: [763428.208053] Code: 00 48 89 df 4d 8d bd 30 01 00 00 e8 17 9a ef ff 49 8b 9d 30 01 00 00 4c 39 fb 0f 84 6f ff ff ff 0f 1f 80 00 00 00 00 44 39 63 28 <78> 1c 44 3b 73 2c 0f 88 58 ff ff ff b9 01 00 00 00 31 d2 4c 89
```

Every time the soft lockup happened at a different instruction of __mptcp_reinject_data, but always in this function.

We have a modified path manager (very similar to mptcp_fm) but haven't touched mptcp_output.c. Any ideas about what happened?

Thanks!

**Comment by zorun** [comment]
You seem to be using an older version, you should try with v0.92.  For instance this is possibly fixed by 04cd2d22d8c832e3bc6f57f2251f288399b56ab9 (which got released in v0.91.2)

**Comment by nadne** [comment]
Oh, thank you very much! I'll try to update my kernel and hope that it fix the problem
